File: Shortest_common_supersequence.py
class Solution:
    def shortestCommonSupersequence(self, X, Y, m, n):
       prev = [0] * (n + 1)
       curr = [0] * (n + 1)
       for i in range(1, m + 1):
           for j in range(1, n + 1):
               if X[i - 1] == Y[j - 1]:
                   curr[j] = 1 + prev[j - 1]
               else:
                   curr[j] = max(curr[j - 1], prev[j])
           prev = curr[:]
       lcs = curr[n]
       return n + m - lcs

File: Shortest_common_supersequence.cpp
class Solution{
    public:
    int shortestCommonSupersequence(string X, string Y, int m, int n){
        vector<int> prev(n + 1, 0), curr(n + 1, 0);
        for(int i = 1; i <= m; i++){
            for(int j = 1; j <= n; j++){
                if(X[i - 1] == Y[j - 1]){
                    curr[j] = 1 + prev[j - 1];
                }else{
                    curr[j] = max(curr[j - 1], prev[j]);
                }
            }
            prev = curr;
        }
        int lcs = curr[n];
        return n + m - lcs;
    }
};
